pycbc/workflow/jobsetup.py

```diff
@@ -960,6 +1114,42 @@ def create_node(self, job_segment, input_file):
                                  tags=self.tags, store_file=self.retain_files)
         return node
 
+class PycbcSplitInspinjExecutable(Executable):
+    """
+    The class responsible for running the pycbc_split_inspinj executable
+    """
+    current_retention_level = Executable.INTERMEDIATE_PRODUCT
+    def __init__(self, cp, exe_name, num_splits, universe=None, ifo=None,
+                 out_dir=None, tags=[]):
+        super(PycbcSplitInspinjExecutable, self).__init__(cp, exe_name,
+                universe, ifo, out_dir, tags)
+        self.out_dir = out_dir
+        self.num_splits = int(num_splits)
+    def create_node(self, parent):
+        node = Node(self)
+
+        node.add_input_opt('--input-file', parent)
+        
+        if parent.name.endswith("gz"):
+            ext = ".xml.gz"
+        else:
+            ext = ".xml"
+
+        out_files = FileList([])
+        for i in range(self.num_splits):
+            curr_tag = 'split%d' % i
+            # FIXME: What should the tags actually be? The job.tags values are
+            #        currently ignored.
```

Comment:
It looks like this FIXME is still present. The tags sent at the __init__ level can never be used as tags should be taken from the parent job. I would fix this be removing the tags kwarg from the __init__ function and supplying tags=[] explicitly when calling super(PycbcSplitInspinjExecutable, self).